panther_gazebo/src/light_converter_node.cpp

```diff
@@ -0,0 +1,138 @@
+// Copyright 2024 Husarion sp. z o.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "panther_gazebo/light_converter_node.hpp"
+
+#include <iostream>
+#include <memory>
+#include <stdexcept>
+
+#include "rclcpp/rclcpp.hpp"
+#include "sensor_msgs/image_encodings.hpp"
+
+namespace panther_gazebo
+{
+
+GZLightConverter::GZLightConverter(
+  const std::string & node_name, const rclcpp::NodeOptions & options)
+: Node(node_name, options)
+{
+  light_pub_ = this->create_publisher<ros_gz_interfaces::msg::Light>("_gz_lights", 1);
+}
+
+void GZLightConverter::Initialize()
+{
+  it_ = std::make_shared<image_transport::ImageTransport>(this->shared_from_this());
+
+  chanel_1_sub_ = std::make_shared<image_transport::Subscriber>(it_->subscribe(
+    "lights/driver/channel_1_frame", 1,
+    [&](const ImageMsg::ConstSharedPtr & msg) { FrameCB(msg, "front_light"); }));
+
+  chanel_2_sub_ = std::make_shared<image_transport::Subscriber>(it_->subscribe(
+    "lights/driver/channel_2_frame", 1,
+    [&](const ImageMsg::ConstSharedPtr & msg) { FrameCB(msg, "rear_light"); }));
+}
+
+void GZLightConverter::FrameCB(const ImageMsg::ConstSharedPtr msg, std::string led_name)
+{
+  std::string warn_msg;
+  if ((this->get_clock()->now() - rclcpp::Time(msg->header.stamp)) > frame_timeout_) {
+    warn_msg = "Timeout exceeded, ignoring frame";
+  } else if (msg->encoding != sensor_msgs::image_encodings::RGBA8) {
+    warn_msg = "Incorrect image encoding ('" + msg->encoding + "')";
+  }
+
+  if (!warn_msg.empty()) {
+    RCLCPP_WARN_STREAM_THROTTLE(this->get_logger(), *this->get_clock(), 5000, warn_msg);
+    return;
+  }
+
+  auto [meanR, meanG, meanB, meanA] = calculateMeanRGBA(msg->data);
```

Comment:
```suggestion
  auto [mean_r, mean_g, mean_b, mean_a] = calculateMeanRGBA(msg->data);
```
applies in the rest of file